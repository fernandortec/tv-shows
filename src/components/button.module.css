button {
  display: flex;
  align-items: center;

  gap: 8px;
  border: none;


  border-radius: 6px;
}

button:hover {
  transition: background-color 0.2s;
  cursor: pointer;
}

.playButton {
  padding: 0.8rem 2rem;

  background: var(--secondary-500);
  color: var(--white);
}

.playButton:hover {
  background-color: var(--secondary-700);
}

.iconButton {
  height: 3.5rem;
  width: 3.5rem;
  padding: 0px;

  display: flex;
  justify-content: center;
  align-items: center;

  background: var(--primary-700);
  color: var(--white);
}

.iconButton:hover {
  background-color: var(--primary-500);
}