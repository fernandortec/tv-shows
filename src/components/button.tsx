import classnames from "classnames";
import { PlayIcon } from "lucide-react";
import type { ButtonHTMLAttributes } from "react";
import styles from "./button.module.css";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
	variant?: "play-now" | "icon-only" | "primary";
}

export function Button({
	variant,
	type = "button",
	...props
}: ButtonProps): JSX.Element {
	return (
		<button
			className={styles.button}
			type={type}
			{...props}
		>
			<PlayIcon fill="white" size="28" />
			Come√ße a ver agora
		</button>
	);
}
