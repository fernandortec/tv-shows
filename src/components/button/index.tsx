import type { ButtonHTMLAttributes } from "react";
import { PlayIcon } from "lucide-react";
import styles from "./styles.module.css";

type Variants = "play-now" | "icon-only" | "primary";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
	variant?: Variants;
}

export function Button({
	variant,
	type = "button",
	children,
	...props
}: ButtonProps): JSX.Element {
	const classNameByVariant: { [key in Variants]: string } = {
		"icon-only": styles.iconButton,
		"play-now": styles.playNowButton,
		primary: styles.primaryButton,
	};

	const currentVariant = variant ?? "primary";

	return (
		<button
			className={classNameByVariant[currentVariant]}
			type={type}
			{...props}
		>
			{variant === "play-now" && <PlayIcon fill="white" size="28" />}
			{children}
		</button>
	);
}
