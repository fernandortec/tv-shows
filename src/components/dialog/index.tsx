import {
	type ReactElement,
	cloneElement,
	useEffect,
	useRef,
	useState,
	type ReactNode,
} from "react";
import styles from "./styles.module.css";
import { X } from "lucide-react";

interface DialogProps {
	children: ReactNode;
	trigger: ReactElement;
}

export function Dialog({ children, trigger }: DialogProps): JSX.Element {
	const [showModal, setShowModal] = useState<boolean>(false);

	const dialogRef = useRef<HTMLDialogElement>(null);

	useEffect(() => {
		if (showModal) {
			return dialogRef.current?.showModal();
		}

		dialogRef.current?.close();
	}, [showModal]);

	function handleOpenModal(): void {
		setShowModal(true);
	}

	function handleCloseModal(): void {
		setShowModal(false);
	}
	return (
		<>
			<div className={styles.trigger}>
				{cloneElement(trigger, { onClick: handleOpenModal })}
			</div>
			<dialog data-modal ref={dialogRef}>
				<button
					type="button"
					className={styles.close}
					onClick={handleCloseModal}
				>
					<X size={24} />
				</button>
				<div className={styles.content}>{children}</div>
			</dialog>
		</>
	);
}
