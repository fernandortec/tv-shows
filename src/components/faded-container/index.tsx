import type { ReactNode } from 'react';
import styles from './styles.module.css';

type Direction = 'top' | 'bottom' | 'both';

export interface FadedContainerProps {
	direction: Direction;
	children: ReactNode;
	containerClassName?: string;
}

export function FadedContainer({
	direction,
	children,
	containerClassName,
}: FadedContainerProps): JSX.Element {
	const classNameBasedOnDirection: { [key in Direction]: string } = {
		both: `${styles.fadeOutFromBottom} ${styles.fadeOutFromTop}`,
		bottom: styles.fadeOutFromBottom,
		top: styles.fadeOutFromTop,
	};

	const className = classNameBasedOnDirection[direction];

	return (
		<div className={containerClassName}>
			<div>
				<div className={className}>{children}</div>
			</div>
		</div>
	);
}
