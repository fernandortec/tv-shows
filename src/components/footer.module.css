.footer {
  display: flex;
  flex-direction: column;

  gap: 8rem;
  padding: 6rem 0px;

  max-width: 80vw;
  margin: auto;
}

.nav {
  display: flex;
  align-items: start;
  justify-content: space-between;
}

.nav p {
  color: white;
  margin-bottom: 0.7rem;
}

.nav ul {
  display: flex;
  flex-direction: column;
  gap: 0.8rem;

  list-style: none;
}

.nav li {
  color: var(--grey-500)
}

.about {
  display: flex;
  justify-content: space-between;

  font-size: 0.8rem;
  color: var(--grey-500)
}

.about div span {
  position: relative;
  margin: 0px 2.5rem;
}