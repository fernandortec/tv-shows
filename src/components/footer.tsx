import styles from "./footer.module.css";

export function Footer(): JSX.Element {
	return (
		<footer className={styles.footer}>
			<nav className={styles.nav}>
				<ul>
					<p>Home</p>
					<li>Categorias</li>
					<li>Dispositivos</li>
					<li>Preços</li>
					<li>FAQ</li>
				</ul>
				<ul>
					<p>Filmes</p>
					<li>Gêneros</li>
					<li>Bombando</li>
					<li>Lançamentos</li>
					<li>Popular</li>
				</ul>
				<ul>
					<p>Séries</p>
					<li>Gêneros</li>
					<li>Bombando</li>
					<li>Lançamentos</li>
					<li>Popular</li>
				</ul>
				<ul>
					<p>Suporte</p>
					<li>Entre em contato</li>	
				</ul>
				<ul>
					<p>Planos</p>
					<li>Planos disponíveis</li>
					<li>Funcionalidades</li>
				</ul>
			</nav>

			<div className={styles.about}>
				<p>@{new Date().getFullYear()} TvShows. Todos direitos reservados</p>

				<div>
					<span>Termos de Uso</span>
					<span>Política de privacidade</span>
					<span>Política de cookies</span>
				</div>
			</div>
		</footer>
	);
}
