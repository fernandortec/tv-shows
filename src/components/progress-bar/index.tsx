import * as Progress from "@radix-ui/react-progress";
import styles from "./styles.module.css";

interface PaginationProgressProps {
	className?: string;
	progressClassName?: string;
	totalValue?: number;
	currentValue?: number;
}

export function PaginationProgress({
	className,
	progressClassName,
	currentValue,
	totalValue,
}: PaginationProgressProps): JSX.Element {
	return (
		<Progress.Root
			className={`${styles.progressRoot} ${className}`}
			value={totalValue}
		>
			<Progress.Indicator
				className={`${styles.progressIndicator} ${progressClassName}`}
				style={{
					transform: `translateX(-${(totalValue ?? 0) - (currentValue ?? 0)}%)`,
				}}
			/>
		</Progress.Root>
	);
}
