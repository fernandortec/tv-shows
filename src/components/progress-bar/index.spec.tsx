import { ProgressBar } from '@/components/progress-bar';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

describe('Progress Bar', () => {
	it('should be all red when fully filled', async () => {
		const wrapper = render(<ProgressBar value={100} />);
    
		expect(wrapper).toMatchSnapshot();
	});
});
