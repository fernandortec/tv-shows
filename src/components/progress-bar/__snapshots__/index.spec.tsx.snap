// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress Bar > should be all red when fully filled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <progress
        class="_progressRoot_37bb2e undefined"
        max="100"
        value="100"
      >
        100
        %
      </progress>
    </div>
  </body>,
  "container": <div>
    <progress
      class="_progressRoot_37bb2e undefined"
      max="100"
      value="100"
    >
      100
      %
    </progress>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
